### PR TITLE
Update libspng, statically link zlib, and allow using zlib-ng

### PR DESCRIPTION
**Pull request by AlexTMjugador**
This rather big PR updates libspng to commit `264476a`, updating the generated bindings, documentation, and bumping the version of the crate accordingly.

It also links against zlib statically, because although that wastes the shared zlib library that may be installed on some systems, this allows following the expectation that Rust executables are as self-contained as possible, and reduces problems in case the end system has an incompatible version of zlib or has no zlib at all.

Also, the `--size_t-is-usize` parameter is added to the `buildgen.sh` script because of relatively recent changes to bindgen; see https://github.com/rust-lang/rust-bindgen/issues/1671.

Finally, the `println!("include=libspng");` call in the `build.rs` script was removed, as it never did anything, because it should begin with the `cargo:` prefix for it to be honored somewhere. See https://kornel.ski/rust-sys-crate#headers for details.

### CODE REVIEW

File: spng-sys/build.rs [hunk 1]
```diff
@@ -13,8 +13,5 @@ fn main() {
     }
     build.compile("spng");
```
Comment (aloucks) [inline spng-sys/build.rs:15]: Please add this back with the updated path. It exposes the spng include directory to downstream cargo crates in the event that you need to depend on spng from another native/sys build.

```rust
    // DEP_SPNG_INCLUDE
    println!("cargo:include=libspng/spng");
```

Edit: You're right, the original output was bugged and needed the `cargo:` prefix.

**Review by aloucks — CHANGES_REQUESTED** [review]
This looks great, thank you. I'll merge and push to crates.io once `build.rs` has been updated.

**Comment by aloucks** [comment]
Could you also add a CHANGELOG entry? Thanks!

**Comment by AlexTMjugador** [comment]
I've just addressed your comments in the latest commits. Thanks!

File: spng-sys/build.rs (hunk 1 above)
Comment (AlexTMjugador) [reply spng-sys/build.rs:15]: Done in the latest commits, thanks.

**Pull request merged**